src/wasm2c.c.tmpl

```diff
@@ -170,6 +170,28 @@ DEFINE_STORE(i64_store32, u32, u64);
 #define I32_TRUNC_U_F64(x) TRUNC_U(u32, f64, 4294967296.,  x)
 #define I64_TRUNC_U_F64(x) TRUNC_U(u64, f64, (f64)UINT64_MAX, x)
 
+#define TRUNC_SAT_S(ut, st, ft, min, smin, minop, max, smax, x) \
+  ((UNLIKELY((x) != (x)))         ? 0                           \
+   : (UNLIKELY(!((x)minop(min)))) ? smin                        \
+   : (UNLIKELY(!((x) < (max))))   ? smax                        \
+                                  : (ut)(st)(x))
+
+#define I32_TRUNC_SAT_S_F32(x) TRUNC_SAT_S(u32, s32, f32, (f32)INT32_MIN, INT32_MIN, >=, 2147483648.f, INT32_MAX, x)
+#define I64_TRUNC_SAT_S_F32(x) TRUNC_SAT_S(u64, s64, f32, (f32)INT64_MIN, INT64_MIN, >=, (f32)INT64_MAX, INT64_MAX, x)
+#define I32_TRUNC_SAT_S_F64(x) TRUNC_SAT_S(u32, s32, f64, -2147483649., INT32_MIN, >, 2147483648., INT32_MAX, x)
+#define I64_TRUNC_SAT_S_F64(x) TRUNC_SAT_S(u64, s64, f64, (f64)INT64_MIN, INT64_MIN, >=, (f64)INT64_MAX, INT64_MAX, x)
+
+#define TRUNC_SAT_U(ut, ft, max, smax, x) \
```

Comment:
what does `s` mean in `smax`? from the code it looks like the integer / non-float maximum, but I'm not able to guess at what `s` is an abbreviation of...

Reply:
saturated was what we were thinking at the time.

Reply:
Oh, I see.. yes, that makes sense!